benchmarks/experimental/sync_batchnorm.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+import tempfile
+import time
+
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+from torch.nn.parallel import DistributedDataParallel as DDP
+
+import fairscale.experimental.nn
+
+
+def benchmark_bn(rank, world_size, init_file, bn_cls):
+    dist.init_process_group(dist.Backend.NCCL, init_method="file://" + init_file, rank=rank, world_size=world_size)
+    x = torch.randn(50, 2048, 7, 7).to(rank)
+    bn = bn_cls(2048).to(rank)
+    bn = DDP(bn, device_ids=[rank])
+    # Warmup
+    for i in range(50):
+        with torch.no_grad():
+            x = bn(x)
+    torch.cuda.synchronize(rank)
+    t0 = time.time()
+    for i in range(100):
+        with torch.no_grad():
+            x = bn(x)
+    torch.cuda.synchronize(rank)
+    t1 = time.time()
+    print("Elapsed time is ", t1 - t0)
```

Comment:
Is this a temporary file or do we intend to add this to our benchmark runs on circleCI?

Reply:
This file is useful for profiling and was helpful to me when iterating on the code. I'm not yet sure we should run this on circleci. But I plan to use it to use it in the future whenever I make changes.

Reply:
just FYI (recent discovery of mine), torch has a `torch.utils.benchmark` util which is very nice for these things, removes some boilerplate code and prettifies the outputs. Example [here](https://github.com/fairinternal/xformers/blob/main/benchmarks/benchmark_core.py). I agree that having these small tools is useful, I've not used that enough myself, should have